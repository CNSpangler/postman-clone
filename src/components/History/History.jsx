import React from 'react';
import PropTypes from 'prop-types';
// import HistoryItem from '../HistoryItem/HistoryItem.jsx';
import ReactJson from 'react-json-view';
import styles from '../App/App.css';

const History = ({ history }) => {
  return (
    <div className={styles.History}>
      <h3>History:</h3>
      <ReactJson src={history} />
    </div>
  );
  // const historyItems = history.map(item => (
  //   <li key={item.id}>
  //     <HistoryItem {...item} />
  //   </li>
  // ));

  // return (
  //   <ul>
  //     <h1>HISTORY</h1>
  //     {historyItems}
  //   </ul>
  // );
};

History.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string,
    method: PropTypes.string,
    body: PropTypes.any
  }))
};

export default History;
