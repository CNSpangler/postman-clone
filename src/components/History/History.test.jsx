import React from 'react';
import { shallow } from 'enzyme';
import History from './History.jsx';

describe('History component', () => {
  it('renders History ', () => {
    const wrapper = shallow(<History />);
    expect(wrapper).toMatchSnapshot();
  });
});
