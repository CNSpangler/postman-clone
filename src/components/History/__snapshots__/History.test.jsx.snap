// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History  1`] = `
<div
  className="History"
>
  <h3>
    History:
  </h3>
  <t
    collapseStringsAfterLength={false}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={true}
    displayObjectSize={true}
    enableClipboard={true}
    groupArraysAfterLength={100}
    iconStyle="triangle"
    indentWidth={4}
    name="root"
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={Object {}}
    style={Object {}}
    theme="rjv-default"
    validationMessage="Validation Error"
  />
</div>
`;
